x/ibc-transfer/keeper/relay.go

```diff
@@ -70,17 +71,41 @@ func (k Keeper) SendTransfer(
 
 	// begin createOutgoingPacket logic
 	// See spec for this logic: https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer#packet-relay
-
 	channelCap, ok := k.scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(sourcePort, sourceChannel))
 	if !ok {
 		return sdkerrors.Wrap(channeltypes.ErrChannelCapabilityNotFound, "module does not own channel capability")
 	}
 
+	// NOTE: denomination and hex hash correctness checked during msg.ValidateBasic
+	prefixedDenom := token.Denom
+
+	// deconstruct the token denomination into the denomination trace info
+	// to determine if the sender is the source chain
+	if strings.HasPrefix(token.Denom, "ibc/") {
+		hexHash := token.Denom[4:]
+		hash, err := types.ParseHexHash(hexHash)
+		if err != nil {
+			return err
+		}
+
+		denomTrace, found := k.GetDenomTrace(ctx, hash)
+		if !found {
+			return sdkerrors.Wrap(types.ErrTraceNotFound, hexHash)
+		}
+		prefixedDenom = denomTrace.GetPrefix() + denomTrace.BaseDenom
+	} else if strings.Contains(token.Denom, "/") {
+		// in the case the user transfers a prefixed denomination,
+		// update the token denomination with the hashed trace info as that's the
+		// representation on the user balance
```

Comment:
```suggestion
		// update the token denomination with the hashed trace info since that is the
		// internally represented denomination on the user balance
```